From 97c2bf127e9682a83a77c3771701f7f12419f0ca Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Wed, 11 Jan 2023 16:23:15 -0500
Subject: [PATCH 87/89] evm: Determine file uid and gid relative to current
 user_ns

To get the correct hash for a file's attributes when in a user namespace,
the uid and gid values have to be calculated relative to the current
user_ns rather the init_user_ns.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/evm/evm_crypto.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/security/integrity/evm/evm_crypto.c b/security/integrity/evm/evm_crypto.c
index 3c1f164e19fb..5716d8d8ac7a 100644
--- a/security/integrity/evm/evm_crypto.c
+++ b/security/integrity/evm/evm_crypto.c
@@ -169,8 +169,8 @@ static void hmac_add_misc(struct evm_namespace *ns,
 	 * filesystem for real on next boot and trust it because
 	 * everything is signed.
 	 */
-	hmac_misc.uid = from_kuid(&init_user_ns, inode->i_uid);
-	hmac_misc.gid = from_kgid(&init_user_ns, inode->i_gid);
+	hmac_misc.uid = from_kuid(current_user_ns(), inode->i_uid);
+	hmac_misc.gid = from_kgid(current_user_ns(), inode->i_gid);
 	hmac_misc.mode = inode->i_mode;
 	crypto_shash_update(desc, (const u8 *)&hmac_misc, sizeof(hmac_misc));
 	if ((ns->evm_hmac_attrs & EVM_ATTR_FSUUID) &&
-- 
2.40.1

